util/include/util/dbutil.h

```diff
@@ -97,6 +97,40 @@ inline std::string getDbDriver()
 #endif
 }
 
+/// @brief Determines if the specified ODB query result only contains
+/// a single entity. That single entity is then stored in 'singleton_'.
+/// @tparam TEntity The type of entities in the query result.
+/// @param result_ The ODB query result in question.
+/// @param singleton_ The variable that receives the first entity (if any).
+/// @return Returns true if 'result_' only contained 'singleton_';
+/// otherwise false.
+template<typename TEntity>
+bool isSingletonResult(odb::result<TEntity>& result_, TEntity& singleton_)
```

Comment:
I think *singleton* and *single result* are two different concepts. This naming is confusing for me.

Reply:
The name originally came from the idea of a 'singleton list' as in a list with only one element.
But it's best to avoid confusion, and `isSingleResult` is also shorter. So I changed this in https://github.com/Ericsson/CodeCompass/pull/736/commits/72c6a6b441d4092364f480c32fe43cb69a23e18e